lib/agris/context.rb

```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 module Agris
   Context = Struct.new(
-    :base_url, :dataset, :datapath, :database, :userid, :password
+    :base_url, :default_dataset, :datapath, :database, :userid, :password
```

Comment:
You are calling this default_dataset but you aren't using it.  Maybe change
https://github.com/westernmilling/agris.rb/pull/29/files#diff-deaf71162770334d0cb52e971509a3cbR30 to

`@dataset || @context.dataset` 

And verify it works as expected in a test?